src/main/java/frc/robot/commands/vision/TestVision.java

```diff
@@ -0,0 +1,96 @@
+package frc.robot.commands.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.Vision;
+
+public class TestVision extends CommandBase{
+  private Drivetrain m_drive;
+  private Vision m_vision;
+  private double m_encoderStart;
+  private Pose2d m_startPose;
+  private double m_encoderPosition;
+  private Pose2d m_currentPose = null;
+  private int m_endCounter = 0;
+  private int m_printCounter=0;
+  private double m_speed;
+
+  //How many frames it has to not see anything to end the command
+  private static final int endDelay = 5;
+
+  //How many frames it will wait between prints
+  private static final int printDelay = 50;
+
+  /**
+   * A command that moves and prints out distances
+   * @param speed What speed to move at
+   * @param drive The drivetrain
+   * @param vision The vision
+   */
+  public TestVision(double speed, Drivetrain drive, Vision vision){
+    addRequirements(drive);
+    m_drive=drive;
+    m_speed=speed;
+    m_vision=vision;
+  }
+
+  private double getDist(){
+    return 
+      m_drive.m_modules[0].getDrivePosition()/4+
+      m_drive.m_modules[1].getDrivePosition()/4+
+      m_drive.m_modules[2].getDrivePosition()/4+
+      m_drive.m_modules[3].getDrivePosition()/4;
+  }
```

Comment:
use the odometry pose rather than individual module encoders

Reply:
you should have a way to temperaraly disable vison inputs to odometry for the test